src/test/ui/suggestions/for-c-in-str.stderr

```diff
@@ -0,0 +1,11 @@
+error[E0277]: the trait bound `&str: std::iter::Iterator` is not satisfied
+  --> $DIR/for-c-in-str.rs:14:14
+   |
+14 |     for c in "asdf" {
+   |              ^^^^^^ `&str` is not an iterator; maybe try calling `.iter()` or a similar method
+   |
+   = help: the trait `std::iter::Iterator` is not implemented for `&str`
+   = note: required by `std::iter::IntoIterator::into_iter`
```

Comment:
it'd be cool if we could highlight the `for`, but this is indeed pretty good still =)

Reply:
I agree. That will require some extra changes to `report_selection_error` in order to figure out that this is part of a `for`-loop desugaring in order to find the leading `for`. For now I think this is enough of an improvement.